src/firebolt/common/util.py

```diff
@@ -37,15 +51,56 @@ def fix_url_schema(url: str) -> str:
     return url if url.startswith("http") else f"https://{url}"
 
 
-def async_to_sync(f: Callable) -> Callable:
+class AsyncJobThread:
+    """
+    Thread runner that allows running async tasks syncronously in a separate thread.
+    Caches loop to be reused in all threads
```

Comment:
Just one question, why do we need a class, that runs a function in another thread, but synchronously? I would add in comments, which exact problem this class is solving